src/transactions/test/TxResultsTests.cpp

```diff
@@ -113,6 +112,236 @@ TEST_CASE("txresults", "[tx][txresults]")
         return ltx.loadHeader().current().scpValue.closeTime;
     };
 
+    // set up world
+    auto root = TestAccount::createRoot(*app);
+    auto nonExisting = TestAccount{*app, getAccount("nonExisting")};
+    auto nonEnoughBalance =
+        root.create("nonEnoughBalance", lm.getLastMinBalance(0));
+    auto enoughBalance = root.create("enoughBalance", startAmount);
+    auto extraSignature = TestAccount{*app, getAccount("extraSignature")};
+
+    auto validateError = [&](TestAccount const& account,
```

Comment:
The list of checks done are in order:
```
no ops -> txMISSING_OPERATION
timebound -> txTOO_EARLY, then txTOO_LATE
no fee -> txINSUFFICIENT_FEE
no account -> txNO_ACCOUNT
seqnum -> txBAD_SEQ
bad auth -> txBAD_AUTH
account balance too low -> txINSUFFICIENT_BALANCE
op check valid -> txFAILED
extra auth -> txBAD_AUTH_EXTRA
```
I would expect this function to instead of taking `authLevel`, would take the place in that list where we expect the failure (type = `TransactionResultCode`).

Then all you need to do is something like this to consolidate all checks:
```c++

        // NB: cases are fall through here as we want to add all possible
       //  negative conditions if we expect a failure at a certain point
        switch (failFromCode)
        {
        case txMISSING_OPERATION:
            rawTx.ops.clear();
        case txTOO_EARLY:
            rawTx.timeBounds.activate().minTime = ct + 1;
        case txTOO_LATE:
            rawTx.timeBounds.activate().maxTime = ct - 1;
        case txINSUFFICIENT_FEE:
            rawTx.fee = 1;
        case txNO_ACCOUNT:
            // handled outside
        case txBAD_SEQ:
            rawTx.seqNum++;
        case txBAD_AUTH:
            badAuth = true;
        //...
        case txBAD_AUTH_EXTRA:
            extraAuth = true;
        };
        // now, sign and check the transaction
        (badAuth ? badAccount : account).sign(tx);
        if (extraAuth)
        {
            account2.sign(tx);
        }
        validateTxResults(tx, *app, validationResult, applyResult);
```
The difference with your version (on top of being more exhaustive) is that we only perform one check with all possible failures added at once (compared to your version that checks for them ~independently); it's actually fine to do this because we want to catch that none of them are tripping (we're using this in a negative context).

Then, the test cases are just things like:
```c++
SECTION("txTOO_LATE")
{
    auto rawTx = nonExisting.rawTx({payment(root, 0)}, 4);
    rawTx.timeBounds.activate().maxTime = ct - 1;
    checkError(rawTx, txINSUFFICIENT_FEE, {1, txTOO_LATE})
}
```

Also, at this layer we don't care to distinguish "no signature" from "bad signature" as what we want to make sure is that "signature verification" is not done before sequence number processing for example.